umake/frameworks/ide.py

```diff
@@ -139,23 +148,76 @@ def get_sha_and_start_download(self, download_result):
             logger.error("Download page changed its syntax or is not parsable (missing sha512)")
             UI.return_main_screen(status_code=1)
         logger.debug("Found download link for {}, checksum: {}".format(url, sha512))
-        self.download_requests.append(DownloadItem(url, Checksum(self.checksum_type, sha512)))
+        self.download_requests.append(DownloadItem(url, Checksum(ChecksumType.sha512, sha512)))
         self.start_download_and_install()
 
     def post_install(self):
         """Create the Eclipse launcher"""
-        icon_filename = "icon.xpm"
-        icon_path = join(self.install_path, icon_filename)
+        DownloadCenter(urls=[DownloadItem(self.icon_url, None)],
+                       on_done=self.save_icon, download=True)
```

Comment:
ah indeed, and it won't be extracted yet so you don't have the destination. Ok, then, let's keep it in post_install for now.